modules/apps/site/site-cms-site-initializer-test/src/testIntegration/java/com/liferay/site/cms/site/initializer/internal/service/test/ObjectEntryLocalServiceTest.java

```diff
@@ -152,47 +152,48 @@ public void testPublishSystemObjectDefinition() throws Exception {
 
-		Role role = _roleLocalService.getRole(
-			systemObjectDefinition.getCompanyId(),
-			RoleConstants.CMS_ADMINISTRATOR);
-
-		String[] actionIds = {
-			ActionKeys.DELETE, ActionKeys.PERMISSIONS, ActionKeys.UPDATE,
-			ActionKeys.VIEW
-		};
-
-		for (String actionId : actionIds) {
-			Assert.assertTrue(
-				_resourcePermissionLocalService.hasResourcePermission(
-					systemObjectDefinition.getCompanyId(),
-					systemObjectDefinition.getClassName(),
-					ResourceConstants.SCOPE_COMPANY,
-					String.valueOf(systemObjectDefinition.getCompanyId()),
-					role.getRoleId(), actionId));
-		}
-
-		String[] viewPermissionRoleNames = {
-			RoleConstants.GUEST, RoleConstants.USER
-		};
-
-		for (String roleName : viewPermissionRoleNames) {
-			role = _roleLocalService.getRole(
-				systemObjectDefinition.getCompanyId(), roleName);
-
-			Assert.assertTrue(
-				_resourcePermissionLocalService.hasResourcePermission(
-					systemObjectDefinition.getCompanyId(),
-					ObjectDefinition.class.getName(),
-					ResourceConstants.SCOPE_INDIVIDUAL,
-					String.valueOf(
-						systemObjectDefinition.getObjectDefinitionId()),
-					role.getRoleId(), ActionKeys.VIEW));
-		}
+		_assertHasResourcePermission(
+			ActionKeys.DELETE, systemObjectDefinition.getClassName(),
+			String.valueOf(_group.getCompanyId()),
+			RoleConstants.CMS_ADMINISTRATOR, ResourceConstants.SCOPE_COMPANY);
+		_assertHasResourcePermission(
+			ActionKeys.PERMISSIONS, systemObjectDefinition.getClassName(),
+			String.valueOf(_group.getCompanyId()),
+			RoleConstants.CMS_ADMINISTRATOR, ResourceConstants.SCOPE_COMPANY);
+		_assertHasResourcePermission(
+			ActionKeys.UPDATE, systemObjectDefinition.getClassName(),
+			String.valueOf(_group.getCompanyId()),
+			RoleConstants.CMS_ADMINISTRATOR, ResourceConstants.SCOPE_COMPANY);
+		_assertHasResourcePermission(
+			ActionKeys.VIEW, systemObjectDefinition.getClassName(),
+			String.valueOf(_group.getCompanyId()),
+			RoleConstants.CMS_ADMINISTRATOR, ResourceConstants.SCOPE_COMPANY);
+		_assertHasResourcePermission(
+			ActionKeys.VIEW, ObjectDefinition.class.getName(),
+			String.valueOf(systemObjectDefinition.getObjectDefinitionId()),
+			RoleConstants.GUEST, ResourceConstants.SCOPE_INDIVIDUAL);
+		_assertHasResourcePermission(
+			ActionKeys.VIEW, ObjectDefinition.class.getName(),
+			String.valueOf(systemObjectDefinition.getObjectDefinitionId()),
+			RoleConstants.USER, ResourceConstants.SCOPE_INDIVIDUAL);
 
 		_objectDefinitionLocalService.deleteObjectDefinition(
 			systemObjectDefinition.getObjectDefinitionId());
 	}
 
+	private void _assertHasResourcePermission(
+			String actionId, String className, String primKey, String roleName,
+			int scope)
+		throws Exception {
+
+		Role role = _roleLocalService.getRole(_group.getCompanyId(), roleName);
+
+		Assert.assertTrue(
+			_resourcePermissionLocalService.hasResourcePermission(
+				_group.getCompanyId(), className, scope, primKey,
+				role.getRoleId(), actionId));
+	}
+
```

Comment:
I think this method is a good candidate for inlining, because it is very close to the internal call to `Assert.assertTrue()`.

What do you think about splitting the two loops into two methods like this:

```java
private void _assertHasResourcePermissionScopeCompany(
		ObjectDefinition objectDefinition, String actionId) {}

private void _assertHasResourcePermissionScopeIndividual(
		ObjectDefinition objectDefinition, String roleName) {}
```